app/controllers/compute_resources_controller.rb

```diff
@@ -42,19 +42,20 @@ def edit
   end
 
   def associate
-    count = 0
-    if @compute_resource.respond_to?(:associated_host)
-      @compute_resource.vms(:eager_loading => true).each do |vm|
-        if Host.for_vm(@compute_resource, vm).empty?
-          host = @compute_resource.associated_host(vm)
-          if host.present?
-            host.associate!(@compute_resource, vm)
-            count += 1
-          end
-        end
+    if @compute_resource.supports_host_association?
+      associator = ComputeResourceHostAssociator.new(@compute_resource)
+      associator.associate_hosts
+      messages = []
+      messages << _('%s VMs were each associated to hosts') % associator.hosts.count
```

Comment:
This does not handle the case for 1 VM (pluralize). Also this is missing a trailing ".".
Can we add a special message when no VM at all was associated?

"No VMs matched any host in Foreman, no VMs were associated."

![image](https://user-images.githubusercontent.com/4107560/34684558-c04ccbf8-f4a5-11e7-8a9c-f9ce532623de.png)